operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -666,13 +666,31 @@ public static void delete(Path key) {
     private static Future<String> getPasswordAsync(SecretOperator secretOperator, String namespace, KafkaClientAuthentication auth) {
         if (auth instanceof KafkaClientAuthenticationPlain) {
             return secretOperator.getAsync(namespace, ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getSecretName())
-                    .compose(secret -> secret == null ? Future.failedFuture("Secret " + ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getSecretName() + " not found") :
-                            Future.succeededFuture(secret.getData().get(((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getPassword())));
+            .compose(secret -> {
+                if (secret == null) {
+                    return Future.failedFuture("Secret " + ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getSecretName() + " not found");
+                } else {
+                    String password = ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getPassword();
```

Comment:
Maybe calling the variable `passwordKey` will make it more readable -> it will be more clear that it is just the key to it and not the password itself.

Reply:
yes, good catch, updated!